extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java

```diff
@@ -45,47 +72,58 @@
     /**
      * Sets realm used by authentication
      */
-    @ConfigItem
+    @ConfigItem(defaultValue = "default")
     Optional<String> authRealm;
 
     /**
      * Sets server name used by authentication
      */
-    @ConfigItem
+    @ConfigItem(defaultValue = "infinispan")
     Optional<String> authServerName;
 
     /**
-     * Sets client subject used by authentication
+     * Sets client subject, necessary for those SASL mechanisms which require it to access client credentials.
      */
     @ConfigItem
     Optional<String> authClientSubject;
 
     /**
-     * Sets callback handler used by authentication
+     * Specifies a {@link CallbackHandler} to be used during the authentication handshake.
+     * The {@link Callback}s that need to be handled are specific to the chosen SASL mechanism.
      */
     @ConfigItem
     Optional<String> authCallbackHandler;
 
+    // @formatter:off
     /**
-     * Sets SASL mechanism used by authentication
+     * Sets SASL mechanism used by authentication.
+     * Available values:
+     * * `DIGEST-MD5` - Uses the MD5 hashing algorithm in addition to nonces to encrypt credentials. This is the default.
+     * * `EXTERNAL` - Uses client certificates to provide valid identities to Infinispan Server and enable encryption.
+     * * `PLAIN` - Sends credentials in plain text (unencrypted) over the wire in a way that is similar to HTTP BASIC
+     *             authentication. You should use `PLAIN` authentication only in combination with TLS encryption.
      */
-    @ConfigItem
+    // @formatter:on
+    @ConfigItem(defaultValue = "DIGEST-MD5")
     Optional<String> saslMechanism;
 
     /**
-     * Sets the trust store path
+     * Specifies the filename of a truststore to use to create the {@link SSLContext}.
+     * You also need to specify a trustStorePassword.
+     * Setting this property also implicitly enables SSL/TLS.
      */
     @ConfigItem
     Optional<String> trustStore;
 
     /**
-     * Sets the trust store password
+     * Specifies the password needed to open the truststore You also need to specify a trustStore.
+     * Setting this property also implicitly enables SSL/TLS.
      */
     @ConfigItem
     Optional<String> trustStorePassword;
 
     /**
-     * Sets the trust store type
+     * Specifies the type of the truststore, such as JKS or JCEKS. Defaults to JKS if trustStore is enabled.
      */
     @ConfigItem
```

Comment:
Should we set the default for the config item as well?

Reply:
i was doubting on this since I don't know if putting a default value here would activate the trustStore

Reply:
I prefer leaving this one